app/javascript/components/rooms/room/join/RoomJoin.jsx

```diff
@@ -88,7 +88,7 @@ export default function RoomJoin() {
     if (hasStarted) {
       toast.success(t('toast.success.room.meeting_started'));
       console.info(`Attempting to join the room(friendly_id): ${friendlyId} meeting in 7s.`);
-      setTimeout(methods.handleSubmit(handleJoin), 7000); // TODO: Improve this race condition handling by the backend.
+      setTimeout(methods.handleSubmit(handleJoin), Math.floor(Math.random() * 4000) + 1000); // Offset join times to ease load on BigBlueButton server
```

Comment:
To prevent all users from hitting the BigBlueButton server at the same time, we can offset the time that they join by making them way a random amount of time (between 1 and 5 seconds).

This is already after atleast 7 seconds of the meeting being started (see below)

Reply:
This was mentioned at the Berlin convention